docs/src/main/asciidoc/_migration-guide-for-4.0.adoc

```diff
@@ -238,6 +238,30 @@ Check link:appendix.html[here] for a full list of configurations.
 
 IMPORTANT: If you're using the new `spring-cloud-azure-starter-active-directory`, you have to specify the `spring.cloud.azure.active-directory.enabled` to true, even if the starter is included in classpath.
 
+.Property mapping from azure-spring-boot-starter-active-directory to spring-cloud-azure-starter-active-directory
+[cols="<30,<~", options="header"]
+|===
+| Legacy properties                                                                        | Morden properties
+| azure.activedirectory.app-id-uri                                                         | spring.cloud.azure.active-directory.app-id-uri
+| azure.activedirectory.application-type                                                   | spring.cloud.azure.active-directory.application-type
+| azure.activedirectory.authorization-clients                                              | spring.cloud.azure.active-directory.authorization-clients
+| azure.activedirectory.authorization-clients.AZURE_CLIENT_NAME.authorization-grant-type   | spring.cloud.azure.active-directory.authorization-clients.AZURE_CLIENT_NAME.authorization-grant-type
+| azure.activedirectory.authorization-clients.AZURE_CLIENT_NAME.on-demand                  | spring.cloud.azure.active-directory.authorization-clients.AZURE_CLIENT_NAME.on-demand
+| azure.activedirectory.authorization-clients.AZURE_CLIENT_NAME.scopes                     | spring.cloud.azure.active-directory.authorization-clients.AZURE_CLIENT_NAME.scopes
+| azure.activedirectory.base-uri                                                           | spring.cloud.azure.active-directory.profile.environment.active-directory-endpoint
+| azure.activedirectory.client-id                                                          | spring.cloud.azure.active-directory.credential.client-id
+| azure.activedirectory.client-secret                                                      | spring.cloud.azure.active-directory.credential.client-secret
+| azure.activedirectory.graph-membership-uri                                               | spring.cloud.azure.active-directory.profile.environment.microsoft-graph-endpoint and spring.cloud.azure.active-directory.user-group.use-transitive-members
```

Comment:
Why is this property mapping to two properties?

Reply:
Because `${azure.activedirectory.graph-membership-uri}` consist of  `${spring.cloud.azure.active-directory.profile.environment.microsoft-graph-endpoint}` and `v1.0/me/transitiveMemberOf or v1.0/me/memberOf`.  `v1.0/me/transitiveMemberOf or v1.0/me/memberOf` can be decided by `${spring.cloud.azure.active-directory.user-group.use-transitive-members}`


Here is related code:

```
    public String getGraphMembershipUri() {
        return getProfile().getEnvironment().getMicrosoftGraphEndpoint()
            + (getUserGroup().getUseTransitiveMembers()
                ? "v1.0/me/transitiveMemberOf"
                : "v1.0/me/memberOf");
    }
```

Reply:
@saragluna ping.

Reply:
I got it that this property don't have a replacement, how about adding a third column and add some description there?

Reply:
@saragluna , updated.